staging/src/k8s.io/apiserver/pkg/util/shufflesharding/shufflesharding.go

```diff
@@ -18,20 +18,37 @@ package shufflesharding
 
 import (
 	"errors"
+	"fmt"
 	"math"
+	"strings"
 )
 
 const maxHashBits = 60
 
-// ValidateParameters can validate parameters for shuffle sharding
-// in a fast but approximate way, including deckSize and handSize
-// Algorithm: maxHashBits >= bits(deckSize^handSize)
-func ValidateParameters(deckSize, handSize int32) bool {
-	if handSize <= 0 || deckSize <= 0 || handSize > deckSize {
-		return false
+// ValidateParameters finds errors in the parameters for shuffle
+// sharding.  Returns an empty slice iff there are no errors.  The
+// entropy requirement is evaluated in a fast but approximate way:
+// bits(deckSize^handSize).
+func ValidateParameters(deckSize, handSize int32) (errs []string) {
```

Comment:
> (errs []string) {

avoid named return value as much as possible, it's conventional in k/k to avoid that usage

Reply:
I can imagine reasons to avoid named return values in general, but in this case I think it works well.  The alternative starts out with
```
 func ValidateParameters(deckSize, handSize int32) []string {
    errs := nil
```
and has more verbosity in the remainder too.